packages/dev/core/src/Physics/v2/physicsConstraint.ts

```diff
@@ -420,3 +422,24 @@ export class PrismaticConstraint extends PhysicsConstraint {
         super(PhysicsConstraintType.PRISMATIC, { pivotA: pivotA, pivotB: pivotB, axisA: axisA, axisB: axisB }, scene);
     }
 }
+
+/**
+ * Creates a SpringConstraint, which is a type of Physics6DoFConstraint. This constraint applies a force at the ends which is proportional
+ * to the distance between ends, and a stiffness and damping factor. The force is calculated as (stiffness * positionError) - (damping * velocity)
+ *
+ * @param pivotA - The first pivot of the constraint in local space.
+ * @param pivotB - The second pivot of the constraint in local space.
+ * @param axisA - The first axis of the constraint in local space.
+ * @param axisB - The second axis of the constraint in local space.
+ * @param minDistance - The minimum distance between the two pivots.
+ * @param maxDistance - The maximum distance between the two pivots.
+ * @param stiffness - The stiffness of the spring.
+ * @param damping - The damping of the spring.
+ * @param scene - The scene the constraint belongs to.
+ * @returns The created SpringConstraint.
+ */
+export class SpringConstraint extends Physics6DoFConstraint {
+    constructor(pivotA: Vector3, pivotB: Vector3, axisA: Vector3, axisB: Vector3, minDistance: number, maxDistance: number, stiffness: number, damping: number, scene: Scene) {
+        super({ pivotA, pivotB, axisA, axisB }, [{ axis: PhysicsConstraintAxis.LINEAR_DISTANCE, minLimit: minDistance, maxLimit: maxDistance, stiffness, damping }], scene);

```

Comment:
Why is the axis limited to `LINEAR_DISTANCE`?

bullet physics's [btGeneric6DofSpringConstraint](https://pybullet.org/Bullet/BulletFull/classbtGeneric6DofSpringConstraint.html) can have separate restrictions on each axis, but is havok impossible?

Reply:
If you want to restrict each axis you can use the Physics6DoFConstraint, the SpringConstraint class is just a straightforward shortcut.

Reply:
I understand. Thank you :D